stages/org.osbuild.rhsm.facts

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python3
+"""
+This stage allows storing artifact-properties of the Red Hat Subscription
+Manager (rhsm-facts) in the built image.
+"""
+
+import json
+import os
+import sys
+
+import osbuild.api
+
+
+SCHEMA = r"""
+"properties": {
+  "facts": {
+    "type": "array",
+    "description": "Array of Red Hat Subscription Manager facts.",
+    "items": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "key": { "type": "string" },
+        "value": { "type": "string" }
+      }
+    }
+  }
```

Comment:
Valid configuration will be like:

```json
{
    "facts": [
        {
            "key": "foo",
            "value": "content1"
        },
        {
            "key": "bar",
            "value": "content2"
        }
    ]
}
```

The same could be achieved with a more simple configuration like:

```json
{
    "facts": {
        "foo": "content1",
        "bar": "content2"
    }
}
```

The latter is much more dense and would use `additionalProperties = True` to implement. It would not allow duplicate keys (probably a good thing), and would not allow easy extensions of the `facts` key, since it would be fully defined already.

I am fine with the initial approach as well. So just something to think about, in case you prefer the second approach as well.

Reply:
I prefer the second approach as well. I believe rhsm facts cannot be nested instead they have some logic to automatically nest dotted keys.